services/cd-service/pkg/repository/repository_test.go

```diff
@@ -1887,3 +1893,95 @@ func TestGitPushDoesntGetStuck(t *testing.T) {
 		})
 	}
 }
+
+type TestWebhookResolver struct {
+	t        *testing.T
+	rec      *httptest.ResponseRecorder
+	requests chan *http.Request
+}
+
+func (resolver TestWebhookResolver) Resolve(insecure bool, req *http.Request) (*http.Response, error) {
+	testhandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		resolver.t.Logf("called with request: %v", *r)
+		resolver.requests <- req
+		close(resolver.requests)
+	})
+	testhandler.ServeHTTP(resolver.rec, req)
+	response := resolver.rec.Result()
+	resolver.t.Logf("responded with: %v", response)
+	return response, nil
+}
+
+func TestSendWebhookToArgoCd(t *testing.T) {
+	tcs := []struct {
+		Name    string
+		Changes TransformerResult
+	}{
+		{
+			Name: "webhook",
+			Changes: TransformerResult{
+				Commits: &CommitIds{
+					Current:  git.NewOidFromBytes([]byte{'C', 'U', 'R', 'R', 'E', 'N', 'T', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}),
+					Previous: git.NewOidFromBytes([]byte{'P', 'R', 'E', 'V', 'I', 'O', 'U', 'S', 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}),
+				},
+			},
+		},
+	}
+	for _, tc := range tcs {
+		tc := tc
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+
+			// given
+			logger, err := zap.NewDevelopment()
+			if err != nil {
+				t.Fatalf("error creating logger: %v", err)
+			}
+			dir := t.TempDir()
+			path := path.Join(dir, "repo")
+			repo, _, err := New2(
+				ctx,
+				RepositoryConfig{
+					URL:  fmt.Sprintf("file://%s", path),
+					Path: path,
+				},
+			)
+			if err != nil {
+				t.Fatalf("new: expected no error, got '%v'", err)
+			}
+			repoInternal := repo.(*repository)
+			repoInternal.config.ArgoWebhookUrl = "http://example.com"
+			rec := httptest.NewRecorder()
+			resolver := TestWebhookResolver{
+				t:        t,
+				rec:      rec,
+				requests: make(chan *http.Request, 1),
+			}
+			repoInternal.config.WebhookResolver = resolver
+
+			// when
+			repoInternal.sendWebhookToArgoCd(ctx, logger, &tc.Changes)
+
+			// then
+			req := <-resolver.requests
+			buf := make([]byte, req.ContentLength)
+			if _, err = io.ReadFull(req.Body, buf); err != nil {
+				t.Errorf("error reading request body: %v", err)
+			}
+			var jsonRequest map[string]any
```

Comment:
let's test the `revision` and/or `htmlUrl` as well, and then I think we're good:
https://github.com/freiheit-com/kuberpult/blob/c6b079a596597288d343d4363b997503874dea38/services/cd-service/pkg/repository/repository.go#L604